src/core/ext/transport/chttp2/client/chttp2_connector.cc

```diff
@@ -171,13 +174,17 @@ void Chttp2Connector::OnHandshakeDone(void* arg, grpc_error_handle error) {
       self->Ref().release();  // Ref held by OnReceiveSettings()
       GRPC_CLOSURE_INIT(&self->on_receive_settings_, OnReceiveSettings, self,
                         grpc_schedule_on_exec_ctx);
-      self->Ref().release();  // Ref held by OnTimeout()
       grpc_chttp2_transport_start_reading(self->result_->transport,
                                           args->read_buffer,
                                           &self->on_receive_settings_, nullptr);
-      GRPC_CLOSURE_INIT(&self->on_timeout_, OnTimeout, self,
-                        grpc_schedule_on_exec_ctx);
-      grpc_timer_init(&self->timer_, self->args_.deadline, &self->on_timeout_);
+      RefCountedPtr<Chttp2Connector> cc = self->Ref();
+      self->event_engine_ = self->args_.channel_args.GetObject<EventEngine>();
```

Comment:
Does it make more sense to store this on `Connect` when the rest of the members are initialized as well?

Reply:
Done, agree it is clearer this way.